packages/atomic/src/components/search/generated-answer/atomic-generated-answer/atomic-generated-answer.tsx

```diff
@@ -201,11 +207,29 @@ export class AtomicGeneratedAnswer implements InitializableComponent {
     return 'mt-0 mb-4 border border-neutral shadow-lg p-6 bg-background rounded-lg p-6 text-on-background';
   }
 
+  private get copyToClipboardTooltip() {
+    if (this.copyError) {
+      return this.bindings.i18n.t('failed-to-copy-generated-answer');
+    }
+
+    return !this.copied
+      ? this.bindings.i18n.t('copy-generated-answer')
+      : this.bindings.i18n.t('generated-answer-copied');
+  }
+
   private async copyToClipboard(answer: string) {
-    await navigator.clipboard.writeText(answer);
-    this.copied = true;
+    try {
+      await navigator.clipboard.writeText(answer);
+      this.copied = true;
+      this.generatedAnswer.logCopyToClipboard();
+    } catch (error) {
+      this.copyError = true;
+      console.error(`Failed to copy to clipboard: ${error}`);
```

Comment:
We should not use `console` directly anywhere in Atomic/Headless as much as possible .

Instead try to use the logger (accessible with `this.bindings.engine.logger` in this case).  

That way the logging stays consistent everywhere/can be formatted by user if they want depending on their environment/can control logging level etc.